### PR TITLE
Added ftp() do ftp end so that users don't have to start/stop the ftp…

### PR DESCRIPTION
**Pull request by samuel-massinon**
… client

@omus Ready to review

### CODE REVIEW

File: src/FTPObject.jl [hunk 1]
```diff
@@ -240,3 +240,15 @@ function ascii(ftp::FTP)
...
+function ftp(code::Function;
+    host::AbstractString="", implicit::Bool=false, ssl::Bool=false,
+    verify::Bool=true, active::Bool=false, user::AbstractString="", pswd::AbstractString="" )
+    ftp_init()
+    ftp_client = FTP(;host=host, implicit=implicit, ssl=ssl, verify=verify, active=active, user=user, pswd=pswd)
+
+    code(ftp_client)
+
+    close(ftp_client)
+    ftp_cleanup()
```
Comment (omus) [inline src/FTPObject.jl]: `code` should be wrapped in a try/catch

File: test/test_ftp_object.jl [hunk 2]
```diff
@@ -246,4 +246,14 @@
 
     end
     ftp_cleanup()
+
+    @testset "ftp() do ftp_client end" begin
+        ftp(ssl=false, user=user, pswd=pswd, host=host) do ftp
+            buff = download(ftp, file_name)
+            actual_buff = readstring(buff)
+            @test actual_buff == file_contents
```
Comment (omus) [inline test/test_ftp_object.jl]: Combine into `@test readstring(buff) == file_contents`

File: test/test_ftp_object.jl [hunk 3]
```diff
@@ -6,18 +6,18 @@
...
+    function no_unexpected_changes(ftp::FTP)
+        @test ftp.ctxt.options.ssl == false
+        @test ftp.ctxt.options.username == user
+        @test ftp.ctxt.options.passwd == pswd
+        @test ftp.ctxt.options.hostname == host
+        @test ftp.ctxt.options.url == "ftp://$host/"
+        @test ftp.ctxt.options.implicit == false
+        @test ftp.ctxt.options.verify_peer == true
+        @test ftp.ctxt.options.active_mode == false
+    end
```
Comment (omus) [inline test/test_ftp_object.jl:12]: Maybe you should implement a `==` for `ftp.ctxt` objects?

File: src/FTPC.jl [hunk 4]
```diff
@@ -477,3 +477,19 @@ Close connection FTP server.
...
+# ISEQUAL
+##############################
+
+function ==(this::RequestOptions, other::RequestOptions)
+    return this.implicit == other.implicit &&
+        this.ssl == other.ssl &&
+        this.verify_peer == other.verify_peer &&
+        this.active_mode == other.active_mode &&
+        this.username == other.username &&
+        this. passwd == other.passwd &&
```
Comment (omus) [inline src/FTPC.jl]: Does this work with the space after the period?

Comment (samuel-massinon) [inline src/FTPC.jl]: Didn't even notice that, but apparently it does.

``` julia
julia> a = "hello"
"hello"

julia> a. data
5-element Array{UInt8,1}:
 0x68
 0x65
 0x6c
 0x6c
 0x6f
```

Comment (samuel-massinon) [inline src/FTPC.jl]: I'll change it regardless.

**Comment by omus** [comment]
LGTM

**Pull request merged**